content/articles/a-sbert-based-flask-web-app-for-automatic-text-summarization/index.md

```diff
@@ -0,0 +1,204 @@
+</head>
+<body>
+    <nav class="navbar navbar-light" style="background-color: #100da1;">
+        <div class="container">
+            <a class="navbar-brand">Summarizer</a>
+        </div>
+    </nav>
+    <form action="/summarize" method="post">
+        <div class="form-group">
+            <label for="exampleFormControlTextarea1" style="padding-top: 2em;">
+                <strong>Enter Your Text Below To Be Summarized:</strong>
+            </label>
+            <br>
+            <textarea class="form-control" id="exampleFormControlTextarea1" rows="10" name="data"></textarea>
+        </div>
+        <br>
+        
+        <button type="submit" class="btn btn-outline-primary">Summarize</button>
+    </form>
+    
+</body>
+</html>
+```
+
+To display a summary of the input text from the `index.html` page, then the `summary.html` file comes in. 
+
+Copy the below html code in to the `summary.html` file:
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Your Summary</title>
+    <style>
+        p{
+            padding-top: 2em;
+            text-align: center;
+            line-height: 3em;           
+            color: black;
+            word-spacing: 0.25em;            
+            font-family: 'Times New Roman', Times, serif;
+        }
+    </style>
+    
+   <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3" crossorigin="anonymous">
+</head>
+<body>
+    <nav class="navbar navbar-light" style="background-color: hsl(236, 96%, 22%);">
+        <div class="container">
+            <a class="navbar-brand">Text SUMMARY</a>
+        </div>
+    </nav>
+    <p>{{ result }}</p>    
+</body>
+</html>
+```
+
+#### Step Three: Back-End coding
+In your project folder, create a file named `app.py` and should contain the below code.
+
+```python
+#importing flask
+from flask import Flask, render_template,request
+#Importing the summarizer
+from summarizer import Summarizer
+from summarizer.sbert import SBertSummarizer
+```
+
+The latest version of `bert-extractive-summarizer` lets you use `Sentence Bert`. It's based on [this](https://arxiv.org/abs/1908.10084) paper and [this]( https://www.sbert.net/) library.
+
+After importing libraries we create create a SBERT model. We'll use our SBERT model to summarize the required content in our next session.
+
+```python
+# Using an instance of SBERT to create the model
+model = SBertSummarizer('paraphrase-MiniLM-L6-v2')# Useful for assignments since it converts phrases and paragraphs into a 384-dimensional dense vector space.

```

Comment:
Can you explain more about this `paraphrase-MiniLM-L6-v2`?

Is it some pre-trained file?